docs/java/how-to-develop-a-workflow-definition-in-java.md

```diff
@@ -146,12 +231,154 @@ Workflow methods annotated with `@QueryMethod` and `@SignalMethod` can be invoke
+Note that the Temporal Java SDK also provides typed `WorkflowStub` versions for these methods.
+
+See the `WorkflowStub.java` reference: <https://github.com/temporalio/sdk-java/blob/master/temporal-sdk/src/main/java/io/temporal/client/WorkflowStub.java>
+
+When using untyped `WorkflowStubs`, we rely on the Workflow Type, Activity Type, Child Workflow Type, as well as Query and Signal names.
+
+See [How to spawn a Workflow Execution in Java](/docs/java/how-to-spawn-a-workflow-execution-in-java).
+
+### Calling other Workflows
+
+Workflows can invoke, and send Signals to, other Workflows purely by name. This helps particularly for executing Workflows written in other language SDKs.
+
+See the [Temporal Polyglot example](https://github.com/tsurdilo/temporal-polyglot) for more information.
+
+### Dynamic Workflows
 
-Temporal uses the [Event Sourcing pattern](https://docs.microsoft.com/en-us/azure/architecture/patterns/event-sourcing) to recover the state of a Workflow object including its threads and local variable values.
-In essence, every time a Workflow state is restored, its code is re-executed from the beginning.
-Note that, during replay, successfully executed Activities are not re-executed because their results are already recorded
-in the Event History of the Workflow Execution.
+Use `DynamicWorkflow` to implement Workflow Types dynamically. When you register a Workflow implementation type that extends `DynamicWorkflow`, it can be used to implement any Workflow type that is not implicitly registered with the Worker.
+
+The main use case for `DynamicWorkflow` is an implementation of custom Domain Specific Languages (DSLs). A single implementation can implement a Workflow Type which by definition is dynamically loaded from some external source.
+You can also use `DynamicWorkflow` when you need a default Workflow that can handle all Workflow Types that are not registered with a Worker.
+
+The Dynamic Workflow interface is implemented with the `execute` method. This method takes in `EncodedValues` that are inputs to the Workflow Execution. These inputs can be specified by the Client when invoking the Workflow Execution.
+
+```Java
+public class MyDynamicWorkflow implements DynamicWorkflow {
+   @Override
+    public Object execute(EncodedValues args) {
+    }
+}
+```
+
+The `DynamicWorkflow` must be registered with a Worker.
+The following rules apply when registering a Dynamic Workflow with a Worker:
+
+- Only one Workflow implementation that extends `DynamicWorkflow` can be registered with a Worker.
+- You can register multiple type-specific Workflow implementations alongside a single `DynamicWorkflow` implementation.
+- Implement the `execute` method for Dynamic Workflow implementations. Do not specify a `@WorkflowMethod` when using Dynamic Workflows.
+- Implement Signals and Queries dynamically.
+
+  ```Java
+        Workflow.registerListener(
```

Comment:
maybe we could add a dynamic query registration here too since we mention it. up to you

Reply:
Yes! I'll add it.

Reply:
Added sample for DynamicQueryHandler. Thanks @tsurdilo!